lib/skate/detours/trip_modification.ex

```diff
@@ -0,0 +1,106 @@
+defmodule Skate.Detours.TripModification do
+  @moduledoc """
+  A module that produces GTFS-flavored TripModification structs for a detour
+
+  More info: https://gtfs.org/realtime/reference/#message-tripmodifications
+  """
+
+  alias Schedule.Gtfs.RoutePattern
+  alias Schedule.Gtfs.Stop
+
+  defmodule Input do
+    @moduledoc """
+    Input for `Skate.Detours.TripModification.for/1`.
+    """
+
+    @type t :: %__MODULE__{
+            route_pattern: RoutePattern.t(),
+            missed_stops: [Stop.t()]
+          }
+
+    defstruct [:route_pattern, :missed_stops]
+  end
+
+  defmodule SelectedTrip do
+    @moduledoc """
+    Struct representing the `selected_trips` part of `Skate.Detours.TripModification`.
+
+    More info: https://gtfs.org/realtime/reference/#message-selectedtrips
+    """
+
+    @type t :: %__MODULE__{
+            trip_ids: [String.t()]
+          }
+
+    defstruct [:trip_ids]
+  end
+
+  defmodule Modification do
+    @moduledoc """
+    Struct representing the `modifications` part of `Skate.Detours.TripModification`.
+
+    More info: https://gtfs.org/realtime/reference/#message-modification
+    """
+
+    @type t :: %__MODULE__{
+            start_stop_selector: String.t(),
+            end_stop_selector: String.t()
+          }
```

Comment:
issue: It seems like this is the wrong type?

From what I see in the [`StopSelector` message docs](https://gtfs.org/realtime/reference/#message-stopselector), `StopSelector` is a object and not a string.

Is there a reason to prefer a string here?

Reply:
Oh - I want to fix this before merging.

Reply:
[Fixed](https://github.com/mbta/skate/pull/2665/commits/5de88c009b263655c61a1bad2dbd7fca1d01d2f0)